flink/src/test/java/org/apache/iceberg/flink/TestIcebergConnector.java

```diff
@@ -0,0 +1,315 @@
+    this.isStreaming = isStreaming;
+  }
+
+  @Override
+  protected TableEnvironment getTableEnv() {
+    if (tEnv == null) {
+      synchronized (this) {
+        if (tEnv == null) {
+          EnvironmentSettings.Builder settingsBuilder = EnvironmentSettings
+              .newInstance()
+              .useBlinkPlanner();
+          if (isStreaming) {
+            settingsBuilder.inStreamingMode();
+            StreamExecutionEnvironment env = StreamExecutionEnvironment
+                .getExecutionEnvironment(MiniClusterResource.DISABLE_CLASSLOADER_CHECK_CONFIG);
+            env.enableCheckpointing(400);
+            env.setMaxParallelism(2);
+            env.setParallelism(2);
+            tEnv = StreamTableEnvironment.create(env, settingsBuilder.build());
+          } else {
+            settingsBuilder.inBatchMode();
+            tEnv = TableEnvironment.create(settingsBuilder.build());
+          }
+          // Set only one parallelism.
+          tEnv.getConfig().getConfiguration()
+              .set(CoreOptions.DEFAULT_PARALLELISM, 1)
+              .set(FlinkTableOptions.TABLE_EXEC_ICEBERG_INFER_SOURCE_PARALLELISM, false);
+        }
+      }
+    }
+    return tEnv;
+  }
+
+  @After
+  public void after() throws TException {
+    sql("DROP TABLE IF EXISTS %s", TABLE_NAME);
+
+    // Clean the created orphan databases and tables from hive-metastore.
+    if (isHiveCatalog()) {
+      HiveMetaStoreClient metaStoreClient = new HiveMetaStoreClient(hiveConf);
+      try {
+        metaStoreClient.dropTable(databaseName(), TABLE_NAME);
+        if (!isDefaultDatabaseName()) {
+          try {
+            metaStoreClient.dropDatabase(databaseName());
+          } catch (Exception ignored) {
+            // Ignore
+          }
+        }
+      } finally {
+        metaStoreClient.close();
+      }
+    }
+  }
+
+  private void testCreateConnectorTable() {
+    Map<String, String> tableProps = createTableProps();
+
+    // Create table under the flink's current database.
+    sql("CREATE TABLE %s (id BIGINT, data STRING) WITH %s", TABLE_NAME, toWithClause(tableProps));
+    sql("INSERT INTO %s VALUES (1, 'AAA'), (2, 'BBB'), (3, 'CCC')", TABLE_NAME);
+    Assert.assertEquals("Should have expected rows",
+        Lists.newArrayList(Row.of(1L, "AAA"), Row.of(2L, "BBB"), Row.of(3L, "CCC")),
+        sql("SELECT * FROM %s", TABLE_NAME));
+
+    FlinkCatalogFactory factory = new FlinkCatalogFactory();
+    Catalog flinkCatalog = factory.createCatalog(catalogName, tableProps, new Configuration());
+    Assert.assertTrue("Should have created the expected database", flinkCatalog.databaseExists(databaseName()));
+    Assert.assertTrue("Should have created the expected table",
+        flinkCatalog.tableExists(new ObjectPath(databaseName(), TABLE_NAME)));
+
+    // Drop and create it again.
+    sql("DROP TABLE %s", TABLE_NAME);
+    sql("CREATE TABLE %s (id BIGINT, data STRING) WITH %s", TABLE_NAME, toWithClause(tableProps));
```

Comment:
Is it possible to omit the schema for cases where the underlying table already exists?

Reply:
Unfortunately, the flink sql don't provide the syntax to support it now,  although I agree it's a better user-experience .   Currently, the flink sql provide the  [CREATE TABLE  a LIKE b](https://iceberg.apache.org/flink/#create-table-like), but it still require that the table b is a flink catalog table.